### PR TITLE
solid cache tweaks

### PR DESCRIPTION
**Pull request by xithan**
also: 
- don't use simplecov if rspec runs for a single file to speed up testing
- include_set can pass options to the included set
- reduce duplication in spec helpers for core and deck testing

### CODE REVIEW

File: wagn/lib/wagn/simplecov_helper.rb [hunk 1]
```diff
@@ -1,60 +0,0 @@
-def card_simplecov_filters
```
Comment (xithan) [inline wagn/lib/wagn/simplecov_helper.rb]: use card's simplecov_helper instead

File: card/spec/support/card_spec_loader.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
+class CardSpecLoader
+  class << self
```
Comment (xithan) [inline card/spec/support/card_spec_loader.rb]: shared code of wagn/lib/wagn/mods_spec_helper.rb and card/spec/spec_helper.rb

File: card/mod/solid_cache/set/abstract/solid_cache.rb [hunk 3]
```diff
@@ -10,31 +10,36 @@
 
 card_accessor :solid_cache, type: :html
 
-format :html do
-  def default_core_args args
-    args[:solid_cache] = true unless args.key?(:solid_cache)
+def self.included host_class
+  host_class.format(host_class.try(:cached_format) || :base) do
```
Comment (ethn) [inline card/mod/solid_cache/set/abstract/solid_cache.rb]: cool solution

File: card/mod/solid_cache/set/abstract/solid_cache.rb [hunk 4]
```diff
@@ -72,21 +77,36 @@ def event_name set, args
...
 
+def generate_content_for_cache changed_card=nil
+  format_type = try(:cached_format) || :base
+  format(format_type)._render_core(solid_cache: false,
+                                   changed_card: changed_card)
+end
+
+def updated_content_for_cache _changed_card=nil
+  generate_content_for_cache
+end
```
Comment (ethn) [inline card/mod/solid_cache/set/abstract/solid_cache.rb]: what's the rationale for separating these?  overrides in WikiRate?

File: card/spec/lib/card/view_cache_spec.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# -*- encoding : utf-8 -*-
```
Comment (ethn) [inline card/spec/lib/card/view_cache_spec.rb]: no objection to these, but they only matter if there are non-ascii characters in the code, right?

File: card/spec/support/card_spec_loader.rb (hunk 2 above)
Comment (ethn) [inline card/spec/support/card_spec_loader.rb]: I like it.

File: card/spec/lib/card/view_cache_spec.rb (hunk 5 above)
Comment (xithan) [inline card/spec/lib/card/view_cache_spec.rb]: I get a bunch of "non-ascii character- default to utf-8" warnings each time I run the tests. Hoped that helped but it didn't.

File: card/mod/solid_cache/set/abstract/solid_cache.rb (hunk 4 above)
Comment (xithan) [inline card/mod/solid_cache/set/abstract/solid_cache.rb]: yes, surgical updates using `changed_card` (in WikiRate) instead of regenerating everything

**Pull request merged**